packages/style-engine/class-wp-style-engine.php

```diff
@@ -242,7 +242,8 @@ public function generate( $block_styles ) {
 			foreach ( $css_rules as $rule => $value ) {
 				$filtered_css = esc_html( safecss_filter_attr( "{$rule}: {$value}" ) );
 				if ( ! empty( $filtered_css ) ) {
-					$css_output .= $filtered_css . '; ';
+					$css_output_suffix = str_ends_with( $filtered_css, ';' ) ? ' ' : '; ';
```

Comment:
Thanks for putting this together @chad1008 🙇 

I'm not convinced this change is required on core. 

Block supports, even before the style engine, has been adding `;` to inline styles. 

See this line here: https://github.com/WordPress/gutenberg/commit/6de16981c7fef3db7cc4a9d6cfcb0ad85c916c62#diff-d61feaa89a4d6e58f54a8f07a04644ea493ef9e203f41a284236abe28a1bd899R86

Furthermore, in between moving from creating inlines styles in block supports files and creating them here in the style engine, the core tests weren't modified and continued to pass.

🤔 

It's a curious one though. I would love to know if this is reproducible on WordPress Core (with no plugins/CSS processing). 

So far I cannot reproduce it.